File: fizzbuzz.pl
% Language: Prolog
% To run: $ gprolog --consult-file fizzbuzz.pl --entry-goal "main"

divs(X,Y) :- 0 is mod(X,Y).
fizzy(X) :- divs(X,3).
buzzy(X) :- divs(X,5).

fizzbuzz(X) :- fizzy(X), buzzy(X) -> print('FizzBuzz');
               fizzy(X) -> print('Fizz');
               buzzy(X) -> print('Buzz');
               R is X, print(R). 

fizzbuzzr(From,To) :- From =< To, 
                        fizzbuzz(From), nl,
                      fizzbuzzr(From+1, To); 
                      true.

main :- fizzbuzzr(1,100), halt.

File: fizzbuzz.go
// Language: Go
// To Run: go run fuzzbuzz.go

package main
import "fmt"

func fizzbuzz(from, to int) {
    for x := from; x <= to; x++ {
        var fizzy, buzzy bool = x % 3 == 0, x % 5 == 0
        if fizzy { fmt.Printf("Fizz") }
        if buzzy { fmt.Printf("Buzz") }
        if !fizzy && !buzzy { fmt.Printf("%d ", x) }
        fmt.Printf("\n")
    }
}

func main() {
    fizzbuzz(1, 100)
}

